2024-05-31 15:05:05,179: asyncio: INFO: [MLTrader] Executing the initialize lifecycle method
2024-05-31 15:05:05,389: alpaca.trading.stream: INFO: connected to: BaseURL.TRADING_STREAM_PAPER
2024-05-31 15:05:05,840: root: INFO: Current backtesting datetime 2023-05-11 10:00:00-04:00
2024-05-31 15:05:05,842: root: INFO: Current backtesting datetime 2023-05-11 11:00:00-04:00
2024-05-31 15:05:05,842: asyncio: INFO: [MLTrader] Executing the before_starting_trading lifecycle method
2024-05-31 15:05:05,842: asyncio: INFO: [MLTrader] [34mExecuting the on_trading_iteration lifecycle method at 2024-05-31 15:05:05[0m
2024-05-31 15:05:05,844: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-05-31 15:05:06,396: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-05-31 15:05:06,492: asyncio: ERROR: [MLTrader] 'Series' object has no attribute 'to_pydatetime'
2024-05-31 15:05:06,498: asyncio: ERROR: [MLTrader] Traceback (most recent call last):
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\strategies\strategy_executor.py", line 923, in run
    self._run_trading_session()
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\strategies\strategy_executor.py", line 871, in _run_trading_session
    self._on_trading_iteration()
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\strategies\strategy_executor.py", line 262, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\strategies\strategy_executor.py", line 286, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\strategies\strategy_executor.py", line 397, in _on_trading_iteration
    raise e
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\strategies\strategy_executor.py", line 369, in _on_trading_iteration
    on_trading_iteration()
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\tools\decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\lumibot\tools\decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\trygv\OneDrive\Dokumenter\IT1\algorithmic-trading-python\finance.py", line 555, in on_trading_iteration
    sentiment = self.get_sentiment()
                ^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\trygv\OneDrive\Dokumenter\IT1\algorithmic-trading-python\finance.py", line 540, in get_sentiment
    today_row = df[df['Date'].to_pydatetime().strftime('%Y-%m-%d %H:%M:%S') == today.strftime('%Y-%m-%d %H:%M:%S')]
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\trygv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\pandas\core\generic.py", line 6299, in __getattr__
    return object.__getattribute__(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Series' object has no attribute 'to_pydatetime'

2024-05-31 15:05:06,500: asyncio: INFO: [MLTrader] Executing the on_bot_crash event method
2024-05-31 15:05:06,511: root: INFO: Backtesting finished
2024-05-31 15:05:06,766: asyncio: WARNING: [MLTrader] Cannot plot returns because the strategy returns are missing
2024-05-31 15:05:06,837: asyncio: WARNING: [MLTrader] Cannot create a tearsheet because the strategy returns are missing
2024-05-31 15:05:06,838: backtest_stats: INFO: Backtest took 0:00:01.659643 for a speed of 0.000
